app/domain/authentication/authn_oidc/fetch_oidc_secrets.rb

```diff
@@ -0,0 +1,58 @@
+require 'command_class'
+require 'conjur/fetch_required_secrets'
+
+module Authentication
+  module AuthnOidc
+
+    # TODO: (later version) Fix CommandClass so we can add errors directly
+    # inside of it
+    #
+    # TODO: list any OIDC or other errors here.  The errors are part of the
+    # API.
+    #
+    # Errors from FetchRequiredSecrets
+    #
+    RequiredResourceMissing = ::Conjur::RequiredResourceMissing
+    RequiredSecretMissing   = ::Conjur::RequiredSecretMissing
+
+    FetchOidcSecrets = CommandClass.new(
+      dependencies: {
+        fetch_secrets: ::Conjur::FetchRequiredSecrets.new
+      },
+      inputs:       %i(conjur_account service_id required_variable_names)
+    ) do
+
+      def call
+        oidc_secrets
+      end
+
+      private
+
+      def required_secrets
+        @required_secrets ||= @fetch_secrets.(resource_ids: required_resource_ids)
+      end
+
+      def required_resource_ids
+        @required_variable_names.map { |var_name| variable_id(var_name) }
+      end
+
+      def variable_id(var_name)
+        "#{@conjur_account}:variable:conjur/authn-oidc/#{@service_id}/#{var_name}"
+      end
+
+      def secret_value(var_name)
+        required_secrets[variable_id(var_name)]
+      end
+
+      def oidc_secrets
```

Comment:
I found this method really hard to follow because it relies on so many complex chained-together helper methods. I know it's generally good practice in Ruby to minimize method size, but I would suggest scaling it back a bit here. I find something like this (which I think does the same thing) easier to follow:

```
def required_resource_ids
  @required_variable_names.map { |var_name| variable_id(var_name) }
end

def variable_id(var_name)
  "#{@conjur_account}:variable:conjur/authn-oidc/#{@service_id}/#{var_name}"
end

def oidc_secrets
  secrets = {}

  required_secrets = @fetch_secrets.(resource_ids: required_resource_ids)
  
  @required_variable_names.each do |variable_name|
    full_var_name = variable_id(var_name)
    secrets[variable_name] = required_secrets[full_var_name]
  end

  secrets
end
```

Reply:
Note that this is no longer memoizing `required_secrets`, but I think you were only doing that because it was being constructed inside the loop, which I don't think is necessary in this rewrite.

Reply:
i agree, this looks clearer